### PR TITLE
Backport of fix(control-plane): acl tokens deleted while pods in graceful shutdown into release/1.2.x

### PR DESCRIPTION
**Pull request by hc-github-team-consul-core**
## Backport

This PR is auto-generated from #3736 to be assessed for backporting due to the inclusion of the label backport/1.2.x.

:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@DanStough
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/consul-k8s/merges: 409 Merge conflict []

The below text is copied from the body of the original PR.

---

### Changes proposed in this PR ###  
- Users have reported that when GracefulShutdown is enabled, ACL tokens are deleted immediately when a pod goes in to TERMINATING, which can disrupt consul DNS and an app's shutdown procedure if that is required.
- The new behavior is that if a pod still exists and has GracefulShutdown enabled, we will skip ACL and node cleanup, instead re-queue the event for the longest shutdown period, plus some buffer to cut down on noise.

### How I've tested this PR ###
- Manual testing in a local kube cluster, deploying an app w/ ACLs and graceful shutdown. I delete a pod and watch the control-plane and consul-dataplane logs to verify when the ACL token is cleaned up.
- Additional unit tests added.

### How I expect reviewers to test this PR ###
👀 ☕ 

### Checklist ###
- [X] Tests added
- [X] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry) 

---

<details>
<summary> Overview of commits </summary>

  - 4a7cfeb036c461912e0db15caabb90dab25615a8 

</details>

### CODE REVIEW

**Comment by DanStough** [comment]
@ndhanushkodi would you remind reviewing the second commit of this PR where we working on the test fix yesterday. Once you give the 👍 I will come back around to the later release branches and make a PR just for the test fix.

File: acceptance/tests/connect/connect_proxy_lifecycle_test.go [hunk 1]
```diff
@@ -206,171 +217,172 @@ func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
...
-}
+//
+//func TestConnectInject_ProxyLifecycleShutdownJob(t *testing.T) {
+//	cfg := suite.Config()
+//
+//	if cfg.EnableTransparentProxy {
+//		// TODO t-eckert: Come back and review this with wise counsel.
+//		t.Skip("Skipping test because transparent proxy is enabled")
+//	}
+//
```
Comment (DanStough) [inline acceptance/tests/connect/connect_proxy_lifecycle_test.go:228]: This was a mistake, fixing!

File: acceptance/tests/connect/connect_proxy_lifecycle_test.go [hunk 2]
```diff
@@ -153,23 +155,29 @@ func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
...
+					case <-gracePeriodTimer.C:
+						break gracePeriodLoop
+					default:
+						output, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), args...)
+						require.NoError(t, err)
+						require.True(t, !strings.Contains(output, "curl: (7) Failed to connect"))
+
+						// If listener draining is enabled, ensure inbound
+						// requests are rejected during grace period.
+						if !drainListenersEnabled {
```
Comment (ndhanushkodi) [inline acceptance/tests/connect/connect_proxy_lifecycle_test.go:172]: The comment says if draining is enabled, connections are rejected but the code is checking if draining listeners is not enabled, that the connection is successful, and I just wanted to confidence-check if that is what we expect

File: acceptance/tests/connect/connect_proxy_lifecycle_test.go [hunk 3]
```diff
@@ -138,7 +139,8 @@ func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
 			require.Len(t, pods.Items, 1)
 			clientPodName := pods.Items[0].Name
 
-			var terminationGracePeriod int64 = 60
+			// We should terminate the pods shortly after envoy gracefully shuts down in our 15s test cases.
+			var terminationGracePeriod int64 = 16
```
Comment (ndhanushkodi) [inline acceptance/tests/connect/connect_proxy_lifecycle_test.go:143]: Maybe we could potentially make this 5 seconds and make the 60 second timeout below something like 30s instead. No worries if that may be more unstable/flaky though

**Review by ndhanushkodi — APPROVED** [review]
The second commit looks good, just two non-blocking questions

File: acceptance/tests/connect/connect_proxy_lifecycle_test.go (hunk 2 above)
Comment (DanStough) [reply acceptance/tests/connect/connect_proxy_lifecycle_test.go:172]: Good catch. I think the code is the expected behavior and I'll update the comment.

File: acceptance/tests/connect/connect_proxy_lifecycle_test.go (hunk 3 above)
Comment (DanStough) [reply acceptance/tests/connect/connect_proxy_lifecycle_test.go:143]: Yeah, I like that. I'll also change the grace period in the tests to 5s.

**Pull request merged**